packages/syft/src/syft/core/node/new/task/oblv_service.py

```diff
@@ -0,0 +1,438 @@
+        "Retrieves the public key present on the Domain Node."
+
+        if len(self.oblv_keys_stash):
+            oblv_keys = self.oblv_keys_stash.get_all()
+            if oblv_keys.is_ok():
+                oblv_keys = oblv_keys.ok()[0]
+            else:
+                return oblv_keys.err()
+
+            public_key_str = (
+                encodebytes(oblv_keys.public_key).decode("UTF-8").replace("\n", "")
+            )
+            return Ok(public_key_str)
+
+        return Err(
+            "Public Key not present for the domain node, Kindly request the admin to create a new one"
+        )
+
+    def get_api_for(
+        self,
+        enclave_metadata: OblvMetadata,
+        signing_key: SyftSigningKey,
+    ):
+        deployment_id = enclave_metadata.deployment_id
+        oblv_client = enclave_metadata.oblv_client
+        if not LOCAL_MODE:
+            if (
+                deployment_id in OBLV_PROCESS_CACHE
+                and OBLV_PROCESS_CACHE[deployment_id][0].poll() is None
+            ):
+                port = OBLV_PROCESS_CACHE[deployment_id][1]
+            else:
+                # randomized port staring point, to quickly find free port
+                port_start = 3000 + random.randint(1, 10_000)  # nosec
+                port = find_available_port(
+                    host="127.0.0.1", port=port_start, search=True
+                )
+            connection_string = f"http://127.0.0.1:{port}"
+        else:
+            port = os.getenv("DOMAIN_CONNECTION_PORT", DOMAIN_CONNECTION_PORT)
+            connection_string = f"http://127.0.0.1:{port}"
+            if not WORKER_MODE:
+                connection_string = connection_string.replace(
+                    "127.0.0.1", "host.docker.internal"
+                )
+
+        req = make_request_to_enclave(
+            connection_string=connection_string + Routes.ROUTE_API.value,
+            deployment_id=deployment_id,
+            oblv_client=oblv_client,
+            oblv_keys_stash=self.oblv_keys_stash,
+            request_method=requests.get,
+            connection_port=port,
+        )
+
+        obj = deserialize(req.content, from_bytes=True)
+        # TODO 🟣 Retrieve of signing key of user after permission  is fully integrated
+        obj.signing_key = signing_key
+        obj.connection = HTTPConnection(connection_string)
+        return cast(SyftAPI, obj)
+
+    @service_method(
+        path="oblv.send_user_code_inputs_to_enclave",
+        name="send_user_code_inputs_to_enclave",
+    )
+    def send_user_code_inputs_to_enclave(
+        self,
+        context: AuthedServiceContext,
+        user_code_id: UID,
+        inputs: Dict,
+        node_name: str,
+    ) -> Result[Ok, Err]:
+        user_code_service = context.node.get_service("usercodeservice")
+        action_service = context.node.get_service("actionservice")
+        user_code = user_code_service.get_by_uid(context, uid=user_code_id)
+
+        res = user_code.status.mutate(
```

Comment:
Should this be a `ObjectMutation` change which gets applied to the `user_code` variable?

Reply:
Here the status object is a `UserCodeStatusContext` class  which is context aware on how to mutate permissions when multiple parties are involved